### PR TITLE
Add change tracking plugin for Services and Packages

### PR DESCRIPTION
**Pull request by niroyb**
Commits will be merged
@Microsoft/omsagent-devs 
@robbiezhang @sw47 
@KrisBash

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/oms-agent-for-linux/53)

### CODE REVIEW

File: installer/conf/omsagent.conf [hunk 1]
```diff
@@ -12,6 +12,25 @@
   run_interval 20m
 </source>
 
+#<source>
```
Comment (robbiezhang) [inline installer/conf/omsagent.conf]: can you create a conf in the omsagent.d for changetracking? ideally, it should be enabled by the change tracking DSC module.

Comment (niroyb) [inline installer/conf/omsagent.conf]: Good point, will do.

**Comment by niroyb** [comment]
I forgot to add these files to the packaging in the install-builder `.data` files.

File: installer/datafiles/base_omsagent.data [hunk 2]
```diff
@@ -105,6 +110,10 @@ fi
 # Folder for omsagent.conf includes
 mkdir -m 755 ${{CONF_DIR}}/omsagent.d 2> /dev/null || true
 
+# Copy over standalone plugin configugarations
```
Comment (sw47) [inline installer/datafiles/base_omsagent.data]: minor: typo

File: source/code/plugins/changetracking_lib.rb [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    def self.prev_hash= (value)
+        @@prev_hash = value
+    end
+
+    def self.instanceXMLtoHash(instanceXML)
+        ret = {}
+        propertyXPath = "PROPERTY"
+        instanceXML.elements.each(propertyXPath) { |inst| 
+            name = inst.attributes['NAME']
+            rexmlText = REXML::XPath.first(inst, 'VALUE').get_text # TODO escape unicode chars like "&amp;"
```
Comment (sw47) [inline source/code/plugins/changetracking_lib.rb]: what will be the behavior if this special character was passed?

File: test/code/plugins/filter_changetracking_test.rb [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+      "Enabled"=> "false",
+      "Controller"=> "init"
+    }
+    instanceXML = ChangeTracking::strToXML(instanceXMLstr).root
+    assert_equal("INSTANCE", instanceXML.name)
+    assert_equal("MSFT_nxServiceResource", instanceXML.attributes['CLASSNAME'])
+    instanceHash = ChangeTracking::serviceXMLtoHash(instanceXML)
+    assert_equal(expectedHash, instanceHash)
+  end
+
```
Comment (sw47) [inline test/code/plugins/filter_changetracking_test.rb]: minor: extra space

File: source/code/plugins/changetracking_lib.rb (hunk 3 above)
Comment (niroyb) [inline source/code/plugins/changetracking_lib.rb]: I unescape the initial xml but the text stays escaped...
The behavior now is to send the escaped character up.
There are only a few cases where that happens.

**Comment by sw47** [comment]
signed off

**Pull request merged**